server/src/main/java/org/apache/kafka/server/config/ShareGroupConfigs.java

```diff
@@ -88,9 +88,9 @@ public class ShareGroupConfigs {
     public static final ConfigDef CONFIG_DEF =  new ConfigDef()
             .defineInternal(SHARE_GROUP_ENABLE_CONFIG, BOOLEAN, SHARE_GROUP_ENABLE_DEFAULT, null, MEDIUM, SHARE_GROUP_ENABLE_DOC)
             .define(SHARE_GROUP_DELIVERY_COUNT_LIMIT_CONFIG, INT, SHARE_GROUP_DELIVERY_COUNT_LIMIT_DEFAULT, between(2, 10), MEDIUM, SHARE_GROUP_DELIVERY_COUNT_LIMIT_DOC)
-            .define(SHARE_GROUP_RECORD_LOCK_DURATION_MS_CONFIG, INT, SHARE_GROUP_RECORD_LOCK_DURATION_MS_DEFAULT, atLeast(1), MEDIUM, SHARE_GROUP_RECORD_LOCK_DURATION_MS_DOC)
-            .define(SHARE_GROUP_MIN_RECORD_LOCK_DURATION_MS_CONFIG, INT, SHARE_GROUP_MIN_RECORD_LOCK_DURATION_MS_DEFAULT, atLeast(1), MEDIUM, SHARE_GROUP_MIN_RECORD_LOCK_DURATION_MS_DOC)
-            .define(SHARE_GROUP_MAX_RECORD_LOCK_DURATION_MS_CONFIG, INT, SHARE_GROUP_MAX_RECORD_LOCK_DURATION_MS_DEFAULT, atLeast(1), MEDIUM, SHARE_GROUP_MAX_RECORD_LOCK_DURATION_MS_DOC)
+            .define(SHARE_GROUP_RECORD_LOCK_DURATION_MS_CONFIG, INT, SHARE_GROUP_RECORD_LOCK_DURATION_MS_DEFAULT, between(1000, 60000), MEDIUM, SHARE_GROUP_RECORD_LOCK_DURATION_MS_DOC)
```

Comment:
we are setting the limit  for SHARE_GROUP_RECORD_LOCK_DURATION_MS_DEFAULT to `between(1000, 60000)`, but at the same time the SHARE_GROUP_MAX_RECORD_LOCK_DURATION_MS_DEFAULT limits are set to `between(30000, 3600000)`. 
SHARE_GROUP_RECORD_LOCK_DURATION_MS_DEFAULT limits wont allow us to change beyond 60000. I think we can set `atLeast(1)` for SHARE_GROUP_RECORD_LOCK_DURATION_MS_DEFAULT and the limits will be taken care by min and max configs.

Reply:
I think they do different things.

`SHARE_GROUP_RECORD_LOCK_DURATION_MS` configures the lock duration for share groups which do not have a specific configuration.

`SHARE_GROUP_MIN_RECORD_LOCK_DURATION_MS` is the minimum value the broker will accept for a group-specific configuration for the lock duration. `SHARE_GROUP_MAX_RECORD_LOCK_DURATION_MS` is the maximum value the broker will accept for a group-specific configuration for the lock duration.

Once we have KAFKA-14511, it will be possible to define the group-specific configurations too.

Reply:
Ok, in that case we need to remove the validations added here https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaConfig.scala#L1147 and update tests in KafkaConfigTest?

Reply:
Thanks. I think one of the checks can be removed because the bounds of the min,max make it unnecessary.

Reply:
Or may be we need to define SHARE_GROUP_RECORD_LOCK_DURATION_MS_CONFIG  config as below, which covers min to max range.

```
.define(SHARE_GROUP_RECORD_LOCK_DURATION_MS_CONFIG, INT, SHARE_GROUP_RECORD_LOCK_DURATION_MS_DEFAULT, between(1000, 3600000), MEDIUM, SHARE_GROUP_RECORD_LOCK_DURATION_MS_DOC)

Reply:
That is not what the KIP says. It says the minimum is 1000ms and the maximum is 60000ms. This applies to share groups which do not have a group-specific configuration for lock duration.